epicmail/app/models/users.py

```diff
@@ -5,132 +5,70 @@
-    def __init__(self, user_id, email, firstname, lastname, password):
-        self.user_id = user_id
+
+    users_list = []
+
+    def __init__(self, email, firstname, lastname, password):
+        self.id = len(User.users_list)+1
         self.firstname = firstname
         self.lastname = lastname
         self.email = email
-        self.password = bcrypt.generate_password_hash(
-            password, app.config.get('BCRYPT_LOG_ROUNDS')
-        ).decode()
+        self.password = password
+        # bcrypt.generate_password_hash(
+        #     password, app.config.get('BCRYPT_LOG_ROUNDS')
+        # ).decode()
         # self.auth_token=auth_token
         self.registered_on = datetime.datetime.now()
 
     def to_dictionary(self):
         return {
-            "id":self.user_id,
+            "id": self.id,
             "firstname":self.firstname,
             "lastname":self.lastname,
             "email":self.email,
             "password":self.password,
             "registered_on":self.registered_on,
             # "token":self.auth_token
         }
-
     @staticmethod
-    def validate_firstname(firstname):
-        """Validation for the field of first name in the User model"""
-        if not firstname:
-            error = {
-                "status": 401,
-                "message":"You must provide the first name field"
-            }
-            return jsonify(error), 401
-        elif len(str(firstname)) < 5:
-            error = {
-                "status": 401,
-                "message":"Names accepted must be at least 5 characters"
-            }
-            return jsonify(error), 401
-        elif not isinstance(firstname, str):
-            error = {
-                "status": 401,
-                "message":"The name must be a string"
-            }
-            return jsonify(error), 401
-        elif re.search("[0-9]", str(firstname)) or re.search("[$#@*&%!~`+=|':;.><,_-]", str(firstname)):
-            error = {
-                "status": 401,
-                "message":"Do not use special characters and numbers on a name"
-            }
-            return jsonify(error), 401
+    def find_user_by_email(email):
+        for user in User.users_list:
+            if email == user.email:
+                return user
 
-    
-    @staticmethod
-    def validate_lastname(lastname):
-        """Validation for the field of last name in the User model"""
-        if not lastname:
-            error = {
-                "status": 401,
-                "message":"You must provide the last name field"
-            }
-            return jsonify(error), 401
-        elif len(str(lastname)) < 5:
-            error = {
-                "status": 401,
-                "message":"Names accepted must be at least 5 characters"
+    def create_user(self,email, firstname, lastname, password):
+        if User.find_user_by_email(self.email):
+            return {
+                "status": 202,
```

Comment:
status code should be 409 please.409 means there is a conflict with the request a user is sending to the server

Reply:
check